src/batch/src/executor/generic_exchange.rs

```diff
@@ -64,7 +64,7 @@ impl CreateSource for DefaultCreateSource {
         let task_output_id = prost_source.get_task_output_id()?;
         let task_id = TaskId::from(task_output_id.get_task_id()?);
 
-        if context.is_local_addr(&peer_addr) {
+        if context.is_local_addr(&peer_addr) && prost_source.local_execute_plan.is_none() {
```

Comment:
What's this?

Reply:
ExchangeSource contains a field called `local_execute_plan` which is used to be executed by remote node. Only `ExchangeSourceImpl::Grpc` can handle this.

Reply:
I think better add it as assert? If peer_addr is local and local_execute_plan is not none, it means some question from our planner?

Reply:
In fact, we can optimize this by avoiding sending a rpc here. We can do this later.

Reply:
When lookup join executor and any probe side executor in the same `CN`. It could happen.